crates/metrics/src/futures/future_tracker.rs

```diff
@@ -13,11 +14,27 @@ use std::time::Instant;
 #[derive(Debug)]
 pub struct ExecutionTime<Output> {
     /// The time spent for real action of the future.
-    pub busy: Duration,
+    busy: Duration,
     /// The idle time of the future.
-    pub idle: Duration,
+    idle: Duration,
     /// The output of the future.
-    pub output: Output,
+    output: Output,
+}
+
+impl<Output> ExecutionTime<Output> {
+    /// Extracts the future output and records the execution report into the metrics.
+    pub fn extract(self, metric: &FuturesMetrics) -> Output {
+        // TODO: Use `u128` when `AtomicU128` is stable.
```

Comment:
nit: we shoould probably have an issue for this

Reply:
Also it might be wise to use something like https://docs.rs/portable-atomic for now